tests/wpunit/PostObjectConnectionQueriesTest.php

```diff
@@ -265,6 +267,38 @@ public function testMaxQueryAmount() {
 		$this->assertTrue( $results['data']['posts']['pageInfo']['hasNextPage'] );
 	}
 
+	public function testPostHasPassword() {
+		// Create a test post with a password
+		$this->createPostObject( [
+			'post_title'    => 'Password protected',
+			'post_type'     => 'post',
+			'post_status'   => 'publish',
+			'has_password'  => true,
+			'post_password' => 'password',
+		] );
+
+		/**
+		 * WP_Query posts with a password
+		 */
+		$wp_query_posts_with_password = new WP_Query( [
```

Comment:
try setting `"posts_per_page" => 1` here

I think there's some other test data that's causing the queries to return more than 1 item, and due to how WPGraphQL cursor pagination works, the default order of a GraphQL query isn't a perfect match with default WP_Query, so I think multiple items are being returned, just not in the correct order?

The other option, would be to not run a WP_Query at all here, but instead do something like: 

```
$graphql_query_posts_with_password = $this->postsQuery( $variables );
$edges = $graphql_query_posts_with_password['data']['posts']['edges'];

foreach ( $edges as $post_edge ) {
   $post = get_post( $post_edge['node']['postId'] );
   $this->assertTrue( ! empty( $post->post_password) );
}
```

This way, even if there is other test data, we still are verifying that only posts with a password set are being returned when using the `hasPassword` argument in a GraphQL request.

Reply:
So, I tried this out this morning and it didn't work. The test is failing b/c posts without a password are in fact getting returned. I'm not sure why, b/c when I query for posts with a password against my own WordPress install on Graphiql, the fix works. Maybe there's something about how I'm setting up the test data that isn't correct?

Reply:
Hmm. Maybe. Let me take another look